### PR TITLE
Issue/271 add auth when getting api data

### PR DESCRIPTION
**Pull request by braddf**
# Pull Request

## Description

Add Auth0 handler to grant auth for API side as well as frontend

Fixes #

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: apps/nowcasting-app/components/helpers/utils.ts [hunk 1]
```diff
@@ -97,6 +99,22 @@ export const getRounded4HoursAgoString = () => {
...
 };
 
+//this is the function I set up that would pass the accessToken from get_token.ts into the Authorization
+//header or the request to the API.
+
+export const axiosFetcherAuth = async (url: string) => {
+  const response = await fetch("/api/get_token");
+  const { accessToken } = await response.json();
+
+  console.log(url);
```
Comment (peterdudfield) [inline apps/nowcasting-app/components/helpers/utils.ts:109]: should we remvoe this logging now?

**Review by peterdudfield — COMMENTED** [review]
remove a console log, but looks good.
If it works locally then I'm happy to test this on vercel

File: apps/nowcasting-app/pages/api/auth/[...auth0].ts [hunk 2]
```diff
@@ -30,7 +30,8 @@ export default withSentry(
 
         await handleLogin(req, res, {
           authorizationParams: {
-            redirect_uri: redirectUri
+            redirect_uri: redirectUri,
+            audience: "https://nowcasting-api-eu-auth0.com/"
```
Comment (peterdudfield) [inline apps/nowcasting-app/pages/api/auth/[...auth0].ts:34]: is this the same for dev and pro?

Comment (braddf) [reply apps/nowcasting-app/pages/api/auth/[...auth0].ts:34]: It won't be I don't think - that was that note to self/team earlier that this needs to live in an Env Var on Vercel before we merge to Production

File: apps/nowcasting-app/components/map/pvLatest.tsx [hunk 3]
```diff
@@ -41,7 +41,7 @@ const useGetForecastsData = (isNormalized: boolean) => {
     axiosFetcherAuth,
     {
       onSuccess: (data) => {
-        setForecastCreationTime(data.forecasts[0].forecastCreationTime);
+        if (data.forecasts?.length) setForecastCreationTime(data.forecasts[0].forecastCreationTime);
```
Comment (rachel-labri-tipton) [inline apps/nowcasting-app/components/map/pvLatest.tsx:44]: just asking out of curiosity why did this `if` statement get added?

Comment (braddf) [reply apps/nowcasting-app/components/map/pvLatest.tsx:44]: As part of the stuff not loading properly, this was causing an error if the data was coming back empty, so just added an extra check. Not perfect still by any means still!

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**